core/admin/mailu/models.py

```diff
@@ -257,18 +257,19 @@ def dns_dmarc_report(self):
     @cached_property
     def dns_autoconfig(self):
         """ return list of auto configuration records (RFC6186) """
+        ports = {port.strip() for port in app.config['PORTS'].split(',')}
         hostname = app.config['HOSTNAME']
         protocols = [
-            ('imap', 143, 20),
-            ('pop3', 110, 20),
-            ('submission', 587, 20),
+            ('imap', 143, 20 if '143' in ports else 0),
+            ('pop3', 110, 20 if '110' in ports else 0),
+            ('submission', 587, 20 if '587' in ports else 0),
         ]
         if app.config['TLS_FLAVOR'] != 'notls':
             protocols.extend([
-                ('autodiscover', 443, 10),
+                ('autodiscover', 443, 10 if '443' in ports else 0),
                 ('submissions', 465, 10),
                 ('imaps', 993, 10),
-                ('pop3s', 995, 10),
+                ('pop3s', 995, 10 if '995' in ports else 0),
```

Comment:
I think there is an `if prio` missing in the loop below (line 278)
`for proto, port, prio in protocols if prio`
for this to work.

Reply:
I don't get why: we always define ``prio`` and that part of the code has not changed

Reply:
What do you want to achive by setting prio to 0?
I think it should lead to the DNS records not being generated. This can be done by add inf "if prio".
Why should we add DNS records for disabled services like pop3 with prio 0?

Reply:
I meant to disable it... but apparently the right way to do that is to change the target. I'll update the PR.